packages/grafana-e2e/src/support/types.ts

```diff
@@ -7,12 +7,28 @@ export type Selectors = Record<string, string | Function>;
 export type PageObjects<S> = { [P in keyof S]: SelectorFunction };
 export type PageFactory<S> = PageObjects<S> & SelectorObject<S>;
 export interface PageFactoryArgs<S extends Selectors> {
-  url?: string;
+  url?: string | Function;
   selectors: S;
 }
 
 export const pageFactory = <S extends Selectors>({ url, selectors }: PageFactoryArgs<S>): PageFactory<S> => {
-  const visit = () => e2e().visit(Url.fromBaseUrl(url));
+  const visit = (args?: string) => {
+    if (!url) {
+      return e2e().visit('');
+    }
+
+    let parsedUrl = '';
+    if (typeof url === 'string') {
+      parsedUrl = Url.fromBaseUrl(url);
+    }
+
+    if (typeof url === 'function' && args) {
+      parsedUrl = Url.fromBaseUrl(url(args));
+    }
+
+    e2e().logToConsole('Visiting', parsedUrl);
+    return e2e().visit(parsedUrl);
```

Comment:
Made it so we can have a param in the url so that we can navigate directly to in this case `dashboard`